kobo/apps/stripe/views.py

```diff
@@ -11,3 +24,63 @@ class ProductViewSet(viewsets.ReadOnlyModelViewSet):
         livemode=settings.STRIPE_LIVE_MODE,
     ).prefetch_related('plan_set')
     serializer_class = ProductSerializer
+
+
+class SubscriptionViewSet(
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+    viewsets.GenericViewSet,
+):
+    model = Subscription
+    serializer_class = SubscriptionSerializer
+    lookup_field = 'id'
+    permission_classes = (IsAuthenticated,)
+
+    def get_queryset(self):
+        user_orgs = self.request.user.organizations_organization.values_list(
```

Comment:
Coercing to a python list is probably resulting in an extra query. When you add tests please make use of `self.assertNumQueries` to ensure we aren't calling extra queries. Be extra careful of lists calling one query per row. In the future subscription history may get complicated and it's a good habit to just always be in.